[css-counter-styles-3] cjk-earthly-branch and cjk-heavenly-stem should fallback to cjk-decimal

vitorroriz opened the pull request:
#### e01c725885a7983395016526cfcd160a6e4999e4
<pre>
[css-counter-styles-3] cjk-earthly-branch and cjk-heavenly-stem should fallback to cjk-decimal
<a href="https://bugs.webkit.org/show_bug.cgi?id=258447">https://bugs.webkit.org/show_bug.cgi?id=258447</a>
rdar://111208503

Reviewed by Tim Nguyen.

cjk-earthly-branch and cjk-heavenly-stem should fallback to cjk-decimal
according to CSSWG resolution: <a href="https://github.com/w3c/csswg-drafts/issues/8975">https://github.com/w3c/csswg-drafts/issues/8975</a>

* LayoutTests/fast/lists/w3-css3-list-styles-fallback-style-expected.txt:
* LayoutTests/fast/lists/w3-css3-list-styles-fallback-style.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-earthly-branch/css3-counter-styles-202-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-earthly-branch/css3-counter-styles-202-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-earthly-branch/css3-counter-styles-202.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-heavenly-stem/css3-counter-styles-205-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-heavenly-stem/css3-counter-styles-205-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-heavenly-stem/css3-counter-styles-205.html:
* Source/WebCore/css/counterStyles.css:
(@counter-style cjk-earthly-branch):
(@counter-style cjk-heavenly-stem):

Canonical link: <a href="https://commits.webkit.org/265474@main">https://commits.webkit.org/265474@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b60885b026820044d7fce75432db393eb67b9cde

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10981 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11194 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11513 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12628 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/10487 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/10997 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13571 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11176 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13416 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11140 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/12035 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/9251 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13034 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9328 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9917 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/17148 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/9297 "Build is in progress. Recent messages:") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10399 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/10070 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13314 "Passed tests") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/10515 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/8601 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9687 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/9808 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2637 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13956 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/10369 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-earthly-branch/css3-counter-styles-202-expected.html:
```diff
@@ -5,38 +5,38 @@
+<meta name="assert" content="list-style-type: cjk-earthly-branch falls back to cjk-decimal after 12 per the spec.">
 <style type='text/css'>
-ol li { list-style-type: cjk-earthly-branch;  }
+ol li { list-style-type: cjk-decimal;  }
```
This is not needed, there's `li` in this file.

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-earthly-branch/css3-counter-styles-202-ref.html:
```diff
@@ -5,38 +5,38 @@
+<meta name="assert" content="list-style-type: cjk-earthly-branch falls back to cjk-decimal after 12 per the spec.">
 <style type='text/css'>
-ol li { list-style-type: cjk-earthly-branch;  }
+ol li { list-style-type: cjk-decimal;  }
```
ditto

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-heavenly-stem/css3-counter-styles-205-expected.html:
```diff
@@ -5,39 +5,39 @@
-<meta name="assert" content="list-style-type: cjk-heavenly-stem produces numbers after 9 per the spec.">
+<meta name="assert" content="list-style-type: cjk-heavenly-stem falls back to cjk-decimal after 10 per the spec.">
 <style type='text/css'>
 ol li { list-style-type: cjk-heavenly-stem;  }
```
ditto

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-heavenly-stem/css3-counter-styles-205-ref.html:
```diff
@@ -5,39 +5,39 @@
-<meta name="assert" content="list-style-type: cjk-heavenly-stem produces numbers after 9 per the spec.">
+<meta name="assert" content="list-style-type: cjk-heavenly-stem falls back to cjk-decimal after 10 per the spec.">
 <style type='text/css'>
 ol li { list-style-type: cjk-heavenly-stem;  }
```
ditto

vitorroriz committed:
[css-counter-styles-3] cjk-earthly-branch and cjk-heavenly-stem should fallback to cjk-decimal
https://bugs.webkit.org/show_bug.cgi?id=258447
rdar://111208503

Reviewed by Tim Nguyen.

cjk-earthly-branch and cjk-heavenly-stem should fallback to cjk-decimal
according to CSSWG resolution: https://github.com/w3c/csswg-drafts/issues/8975

* LayoutTests/fast/lists/w3-css3-list-styles-fallback-style-expected.txt:
* LayoutTests/fast/lists/w3-css3-list-styles-fallback-style.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-earthly-branch/css3-counter-styles-202-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-earthly-branch/css3-counter-styles-202-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-earthly-branch/css3-counter-styles-202.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-heavenly-stem/css3-counter-styles-205-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-heavenly-stem/css3-counter-styles-205-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-counter-styles/cjk-heavenly-stem/css3-counter-styles-205.html:
* Source/WebCore/css/counterStyles.css:
(@counter-style cjk-earthly-branch):
(@counter-style cjk-heavenly-stem):

Canonical link: https://commits.webkit.org/265474@main
